### PR TITLE
Added style checker

### PR DESCRIPTION
**Pull request by jasonzliang**
I added  a tool under tests called check_style.sh to show any pep8 style violations.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -29,3 +29,4 @@ google-cloud-pubsub
 oauth2client==3.0.0
 
 boto3
+pep8
```
Comment (karlmutch) [inline requirements.txt]: This would not be needed in production ?  Can we do a manual pip install inside the script instead ?

Comment (karlmutch) [reply requirements.txt]: How and when will check style get run ?

Comment (pzhokhov) [reply requirements.txt]: I use flake8 for checkstyle; we can make it a part of tests, for example - or as a separate task in CI

Comment (pzhokhov) [reply requirements.txt]: if it is a separate task, we don't necessarily need it in requirements.txt for the project, and we can install it only for the travis job

**Comment by jasonzliang** [comment]
I have added the packages required for testing to test_requirements.txt

**Comment by pzhokhov** [comment]
So how is checkstyle being run in this case ? Is it triggered automatically when you do pip install?

**Comment by jasonzliang** [comment]
You need to run check_style.sh manually.

**Comment by pzhokhov** [comment]
hang on, but when are the dependencies from test_requirements.txt are being installed? Do you need to install them manually as well?

**Comment by jasonzliang** [comment]
when you do python setup.py test, it will automatically install these packages for you

**Pull request merged**